### PR TITLE
Adding checkboxes to hide comments

### PR DESCRIPTION
**Pull request by Astra3**
I've added checkboxes to source code view for temporarily hiding comments.

https://github.com/mrlvsb/kelvin/assets/18398260/d7f11ed5-0329-4fe7-8a5e-b78343b601ee

### CODE REVIEW

**Comment by geordi** [comment]
Thanks for nice addition! However, I have a small problem with the "positive/negative" thinking about this problem. Could you, please, change the logic to "show comments" and make them visible by default? I think if makes things easier to reason about. Thank you.

**Comment by Kobzol** [comment]
Thanks for looking at this :) I wonder if we could integrate this in a better way with the button on the right side that already hides all comments. The obvious thing would be to add another button, but maybe we could make the existing button tri-state? "Show all (default) / Hide automated / Hide all".

**Comment by Astra3** [comment]
Having a tristate button actually sounds better to me than the first option, I'll try to look into it with also some icons to communicate the changes.

**Comment by Astra3** [comment]
I've pushed a commit where the comment button is now *that* button.

File: frontend/src/stores.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
...
-export const hideComments = writable({
-    automated: false,
-    student_teacher: false,
-})
+export const HideCommentsState = {
+    NONE: "none",
+    AUTOMATED: "automated",
+    ALL: "all",
+}
+export const hideComments = writable(HideCommentsState.NONE)
```
[the remark this answers is not in the document]
Comment (Astra3) [reply frontend/src/stores.js:8]: I also think so, but I spent about an hour trying to do it differently (filtering the comments passed to the component) and it just wasn't worth it, wouldn't recommend.

**Pull request merged**